virtcontainers/qemu.go

```diff
@@ -1051,33 +1058,64 @@ func (q *qemu) hotplugMemory(memDev *memoryDevice, op operation) (int, error) {
 		return 0, fmt.Errorf("cannot hotplug negative size (%d) memory", memDev.sizeMB)
 	}
 
-	// We do not support memory hot unplug.
-	if op == removeDevice {
-		// Dont fail for now, until we fix it.
-		// We return that we only unplugged 0
-		q.Logger().Warn("hot-remove VM memory not supported")
-		return 0, nil
-	}
-
+	q.Logger().Debugf("hotplug %dMB mem", memDev.sizeMB)
 	err := q.qmpSetup()
 	if err != nil {
 		return 0, err
 	}
 
-	maxMem, err := q.hostMemMB()
-	if err != nil {
-		return 0, err
-	}
-
-	// calculate current memory
 	currentMemory := int(q.config.MemorySize) + q.state.HotpluggedMemory
 
-	// Don't exceed the maximum amount of memory
-	if currentMemory+memDev.sizeMB > int(maxMem) {
-		return 0, fmt.Errorf("Unable to hotplug %d MiB memory, the SB has %d MiB and the maximum amount is %d MiB",
-			memDev.sizeMB, currentMemory, q.config.MemorySize)
+	if memDev.sizeMB == 0 {
+		q.Logger().WithFields(
```

Comment:
Nit: this could be simplified to:

```suggestion
		q.Logger().WithField("hotplug", "memory").Debug("updating balloon as hotplug not required")
```